Jenkinsfile

```diff
@@ -25,52 +12,40 @@ stage('Checkout') {
 stage('Build & Test') {
     parallel(
         backend: {
-            node('docker') {
+            node('superstellar-docker-17.12') {
                 withCleanup {
                     unstash 'source'
 
-                    docker.image('golang:1.7.1').inside("-e HOME=/go -w /go/src/superstellar -v ${pwd()}:/go/src/superstellar") {
-                        sh 'git config --global user.name "Dummy" && git config --global user.email "dummy@example.com"'
-                        sh """
-                            go get superstellar github.com/onsi/ginkgo github.com/onsi/gomega
-                            go build superstellar
-                            go test superstellar/...
-                        """
-                        sh 'cp /go/bin/superstellar .'
-                    }
+                    sh 'docker build -t superstellar-backend-builder:latest --target builder -f docker/backend/Dockerfile .'
```

Comment:
This label is risky. What if 2 builds run at once? Labeling with `latest` should happen within synchronized block.

```
stage(name: "testing", concurrency: 1) {
}
```
I think you should build the container and allow parallel builds but tag with e.g. build number of uuid.
Then after build succeeds in the next stage limit concurrency and tag with `latest`.